meta/hash.go

```diff
@@ -123,6 +123,16 @@ func AlreadyReconciled(o interface{}) (bool, error) {
 	return observedGeneration.MatchGeneration(generation), nil
 }
 
+func EqualStatus(old, new interface{}) bool {
+	oldStatus, oldExists := extractStatusFromObject(old)
+	newStatus, newExists := extractStatusFromObject(new)
+	if oldExists && newExists {
+		return reflect.DeepEqual(oldStatus, newStatus)
+	} else {
+		return !oldExists && !newExists
```

Comment:
What if oldObject does not have anything in status but newObject has something?

Reply:
! equal